src/canvas.rs

```diff
@@ -502,12 +502,24 @@ impl Painter {
                     }
                 };
 
+                let mut mem_rows = 2;
+                if cfg!(feature = "zfs")
+                    && (cfg!(target_os = "linux") || cfg!(target_os = "freebsd"))
+                {
+                    let arc_data: &[(f64, f64)] = &app_state.converted_data.arc_data;
+                    if let Some(arc) = arc_data.last() {
+                        if arc.1 != 0.0 {
+                            mem_rows += 1;
+                        }
+                    }
+                }
```

Comment:
nit: could you just use a single `#[cfg]` block to enable/disable this on compilation, and set `mem_rows` that way? IIRC `if cfg!` does the check on runtime but that's not really needed.

Reply:
I changed the block to a single `#[cfg]` block. 
I would like to preserve runtime mutability if possible of `mem_rows`.
So if in the future more features get added (ex. video ram from `gpu` feature flag) they can be added in a similar style of only mutating the layout when valid data is detected at runtime.